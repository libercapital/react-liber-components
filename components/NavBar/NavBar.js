"use strict";

require("core-js/modules/es6.object.define-property");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.string.iterator");

require("core-js/modules/es6.weak-map");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

require("core-js/modules/es6.regexp.to-string");

require("core-js/modules/es6.date.to-string");

require("core-js/modules/es6.object.to-string");

require("core-js/modules/es7.symbol.async-iterator");

require("core-js/modules/es6.symbol");

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.is-array");

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _DropDown = require("../DropDown");

var _Util = require("../Util");

var _NavBar = require("./NavBar.styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance"); }

function _iterableToArrayLimit(arr, i) { if (!(Symbol.iterator in Object(arr) || Object.prototype.toString.call(arr) === "[object Arguments]")) { return; } var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var button = function button(menu) {
  var _menu$icon = menu.icon,
      icon = _menu$icon === void 0 ? null : _menu$icon,
      label = menu.label,
      _menu$sublabel = menu.sublabel,
      sublabel = _menu$sublabel === void 0 ? null : _menu$sublabel;
  return _react["default"].createElement(_NavBar.Label, null, _react["default"].createElement("li", null, icon, label), sublabel && _react["default"].createElement("li", null, _react["default"].createElement("span", null, sublabel)));
};

var NavBar = function NavBar(_ref) {
  var children = _ref.children,
      menu = _ref.menu,
      contentSize = _ref.contentSize,
      window = _ref.window;
  var content = menu.content;

  var _useState = (0, _react.useState)(window ? (0, _Util.getScreenSize)(window) : 'large'),
      _useState2 = _slicedToArray(_useState, 2),
      screenSize = _useState2[0],
      setScreenSize = _useState2[1];

  (0, _react.useEffect)(function () {
    if (window) {
      var handleResize = function handleResize() {
        setScreenSize((0, _Util.getScreenSize)(window));
      };

      window.addEventListener('resize', handleResize);
      return function () {
        window.removeEventListener('resize', handleResize);
      };
    }

    return undefined;
  }, window);
  var small = screenSize === 'small' || screenSize === 'medium';
  return _react["default"].createElement(_NavBar.Nav, null, _react["default"].createElement(_NavBar.Content, {
    size: contentSize
  }, _react["default"].createElement(_NavBar.Logo, null), children, _react["default"].createElement(_NavBar.Menu, null, _react["default"].createElement(_DropDown.SimpleDropDown, {
    position: "bottom-left",
    label: !small ? button(menu) : undefined,
    customButton: small ? _react["default"].createElement(_NavBar.Sandwich, null) : undefined
  }, content))));
};

NavBar.propTypes = {
  children: _propTypes["default"].oneOfType([_propTypes["default"].any]),
  menu: _propTypes["default"].shape({
    icon: _propTypes["default"].node,
    label: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].string]),
    sublabel: _propTypes["default"].string,
    content: _propTypes["default"].oneOfType([_propTypes["default"].any])
  }),
  contentSize: _propTypes["default"].string,
  window: _propTypes["default"].objectOf(_propTypes["default"].any)
};
NavBar.defaultProps = {
  children: null,
  menu: {},
  contentSize: null,
  window: null
};
var _default = NavBar;
exports["default"] = _default;