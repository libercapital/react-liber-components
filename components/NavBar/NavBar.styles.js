"use strict";

require("core-js/modules/es6.object.define-property");

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.object.to-string");

require("core-js/modules/es6.string.iterator");

require("core-js/modules/es6.weak-map");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Sandwich = exports.Label = exports.Menu = exports.Logo = exports.Content = exports.Nav = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _Logo = require("../Logo");

var _Icons = require("../Icons");

var _Themes = require("../Themes");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var getTheme = function getTheme(value, fallback) {
  return (0, _styledComponents.css)(["", ""], function (props) {
    return (0, _Themes.handleThemeFromObject)(props.theme, value, fallback);
  });
};

var Nav = _styledComponents["default"].nav.withConfig({
  displayName: "NavBarstyles__Nav",
  componentId: "sc-1m4mju1-0"
})(["width:100vw;max-width:100%;min-height:96px;max-height:96px;background-color:", ";margin-bottom:12px;display:flex;justify-content:center;align-items:center;"], getTheme('colors.white', '#ffffff'));

exports.Nav = Nav;

var Content = _styledComponents["default"].div.withConfig({
  displayName: "NavBarstyles__Content",
  componentId: "sc-1m4mju1-1"
})(["height:100%;width:", ";max-width:100%;display:flex;justify-content:flex-start;align-items:center;"], function (props) {
  return props.width || '1280px';
});

exports.Content = Content;
var Logo = (0, _styledComponents["default"])(_Logo.Logo).withConfig({
  displayName: "NavBarstyles__Logo",
  componentId: "sc-1m4mju1-2"
})(["--size:96px;min-width:var(--size);min-height:var(--size);max-width:var(--size);max-height:var(--size);"]);
exports.Logo = Logo;

var Menu = _styledComponents["default"].div.withConfig({
  displayName: "NavBarstyles__Menu",
  componentId: "sc-1m4mju1-3"
})(["margin-left:auto;margin-right:32px;display:flex;justify-content:center;align-items:center;position:relative;"]);

exports.Menu = Menu;

var Label = _styledComponents["default"].ul.withConfig({
  displayName: "NavBarstyles__Label",
  componentId: "sc-1m4mju1-4"
})(["font-family:Roboto;font-size:16px;font-weight:500;line-height:22px;text-align:right;color:#405f71;margin:0px;list-style-type:none;& > li > svg{fill:#798e9b;}& > li > span{font-family:Roboto;font-size:12px;line-height:17px;letter-spacing:0px;text-align:right;color:#b6c2c9;}"]);

exports.Label = Label;
var Sandwich = (0, _styledComponents["default"])(_Icons.MdMenu).withConfig({
  displayName: "NavBarstyles__Sandwich",
  componentId: "sc-1m4mju1-5"
})(["--size:32px;min-width:var(--size);min-height:var(--size);max-width:var(--size);max-height:var(--size);fill:#798e9b;"]);
exports.Sandwich = Sandwich;