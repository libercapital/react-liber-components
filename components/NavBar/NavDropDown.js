'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Util = require('../Util');

var _Util2 = _interopRequireDefault(_Util);

var _styles = require('./styles');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function validateItem(props, propName, componentName) {
  var children = props[propName];
  if (children.constructor !== Array) {
    children = Array(children);
  }
  children.forEach(function (child) {
    if (child.type !== _styles.NavDropDownItem) {
      return new Error(componentName + ' only accept NavDropDownItem as children.');
    }
    return null;
  });
  return null;
}

var NavDropDown = function (_Component) {
  _inherits(NavDropDown, _Component);

  function NavDropDown() {
    _classCallCheck(this, NavDropDown);

    var _this = _possibleConstructorReturn(this, (NavDropDown.__proto__ || Object.getPrototypeOf(NavDropDown)).call(this));

    _this.handleClick = function () {
      _this.setState(function (prevState) {
        return { show: !prevState.show };
      });
    };

    _this.state = {
      show: false
    };
    return _this;
  }

  _createClass(NavDropDown, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          _styles.NavLink,
          {
            active: this.props.active,
            onClick: this.handleClick,
            arrow: true
          },
          this.props.label
        ),
        this.state.show ? _react2.default.createElement(
          _Util2.default,
          { onOutsideClick: this.handleClick },
          _react2.default.createElement(
            _styles.NavDropDownMenu,
            null,
            this.props.children
          )
        ) : null
      );
    }
  }]);

  return NavDropDown;
}(_react.Component);

NavDropDown.propTypes = {
  children: validateItem,
  active: _propTypes2.default.bool,
  label: _propTypes2.default.string.isRequired
};
NavDropDown.defaultProps = {
  active: false,
  children: null
};
exports.default = NavDropDown;