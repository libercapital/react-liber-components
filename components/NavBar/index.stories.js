"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.navBar = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _NavBar = _interopRequireDefault(require("./NavBar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Viewport = _styledComponents["default"].div.withConfig({
  displayName: "indexstories__Viewport",
  componentId: "sc-99m15p-0"
})(["width:100vw;height:100vh;background-color:#fafdff;"]);

var _default = {
  title: 'Components 2.0|NavBar',
  parameters: {
    jest: ['NavBar']
  }
};
exports["default"] = _default;

var navBar = function navBar() {
  return _react["default"].createElement(Viewport, null, _react["default"].createElement(_NavBar["default"], {
    menu: {
      label: 'Nome do usu√°rio',
      sublabel: 'Nome da empresa'
    }
  }));
};

exports.navBar = navBar;
navBar.story = {
  name: 'NavBar'
};