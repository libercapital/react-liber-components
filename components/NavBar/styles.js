'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MrAuto = exports.LoggedIn = exports.DropDownSeparator = exports.NavDropDownMenu = exports.NavDropDownLink = exports.NavLink = exports.ResponsiveNavDropDownItem = exports.NavDropDownItem = exports.ResponsiveNavItem = exports.NavItem = exports.NavList = exports.NavLogo = exports.Nav = exports.UserIcon = undefined;

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _Logo = require('../Logo');

var _Icons = require('../Icons');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var UserIcon = exports.UserIcon = (0, _styledComponents2.default)(_Icons.FaUser).withConfig({
  displayName: 'styles__UserIcon',
  componentId: 'e9mwi9-0'
})(['width:15px;height:15px;margin-right:5px;']);

var Nav = exports.Nav = _styledComponents2.default.nav.withConfig({
  displayName: 'styles__Nav',
  componentId: 'e9mwi9-1'
})(['position:relative;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:column;background:white;padding:0;margin-bottom:15px;']);

var NavLogo = exports.NavLogo = (0, _styledComponents2.default)(_Logo.Logo).withConfig({
  displayName: 'styles__NavLogo',
  componentId: 'e9mwi9-2'
})(['margin-right:28.8px;display:inline-flex;']);

var NavList = exports.NavList = _styledComponents2.default.ul.withConfig({
  displayName: 'styles__NavList',
  componentId: 'e9mwi9-3'
})(['display:flex;padding-left:0;list-style:none;margin:0;margin-top:15px;@media (min-width:576px){margin-top:15px;}@media (min-width:768px){margin-top:15px;}@media (min-width:992px){margin-top:0px;}@media (min-width:1200px){margin-top:0px}']);

var NavItem = exports.NavItem = _styledComponents2.default.li.withConfig({
  displayName: 'styles__NavItem',
  componentId: 'e9mwi9-4'
})(['margin:0px 6px;position:relative;display:none;&[role=separator]{margin:initial;height:1px;margin:0.5rem 0;overflow:hidden;background-color:#eceeef;}@media (min-width:576px){display:none;}@media (min-width:768px){display:none;}@media (min-width:992px){display:inherit;}@media (min-width:1200px){display:inherit;}']);
var ResponsiveNavItem = exports.ResponsiveNavItem = (0, _styledComponents2.default)(NavItem).withConfig({
  displayName: 'styles__ResponsiveNavItem',
  componentId: 'e9mwi9-5'
})(['display:inherit;@media (min-width:576px){display:inherit;}@media (min-width:768px){display:inherit;}']);

var NavDropDownItem = exports.NavDropDownItem = (0, _styledComponents2.default)(ResponsiveNavItem).withConfig({
  displayName: 'styles__NavDropDownItem',
  componentId: 'e9mwi9-6'
})(['margin:0px;']);

var ResponsiveNavDropDownItem = exports.ResponsiveNavDropDownItem = (0, _styledComponents2.default)(NavDropDownItem).withConfig({
  displayName: 'styles__ResponsiveNavDropDownItem',
  componentId: 'e9mwi9-7'
})(['@media (min-width:576px){display:inherit;}@media (min-width:768px){display:inherit;}@media (min-width:992px){display:none;}@media (min-width:1200px){display:none;}']);

var NavLink = exports.NavLink = _styledComponents2.default.a.withConfig({
  displayName: 'styles__NavLink',
  componentId: 'e9mwi9-8'
})(['font-size:17.1px;font-weight:200;text-align:left;color:', ';display:block;padding:36px 10.8px;padding-left:', ';cursor:pointer;&:hover{color:#009dff;}&:after{display:', ';width:0;height:0;margin-left:6px;vertical-align:middle;content:"";border-top:0.3em solid;border-right:0.3em solid transparent;border-left:0.3em solid transparent;}'], function (props) {
  return props.active ? '#009dff' : '#888888';
}, function (props) {
  return props.leftPadding ? props.leftPadding + '!important' : '0.6rem';
}, function (props) {
  return props.arrow ? 'inline-block' : 'none';
});
var NavDropDownLink = exports.NavDropDownLink = (0, _styledComponents2.default)(NavLink).withConfig({
  displayName: 'styles__NavDropDownLink',
  componentId: 'e9mwi9-9'
})(['padding:0;margin:0 1rem;display:block;line-height:2.5rem;']);

var NavDropDownMenu = exports.NavDropDownMenu = _styledComponents2.default.ul.withConfig({
  displayName: 'styles__NavDropDownMenu',
  componentId: 'e9mwi9-10'
})(['position:absolute;top:100%;left:auto;right:0;z-index:1000;float:left;min-width:200px;padding:0.5rem 0;margin:0.125rem 0 0;font-size:1rem;color:#292b2c;font-weight:400;text-align:left;list-style:none;background-color:#fff;background-clip:padding-box;border:1px solid rgba(0,0,0,0.15);border-radius:0.25rem;border-bottom:4px solid #009DFF;font-family:Roboto;']);

var DropDownSeparator = exports.DropDownSeparator = _styledComponents2.default.li.withConfig({
  displayName: 'styles__DropDownSeparator',
  componentId: 'e9mwi9-11'
})(['height:1px;margin:0.5rem 0;overflow:hidden;background-color:#eceeef;']);

var LoggedIn = exports.LoggedIn = _styledComponents2.default.div.withConfig({
  displayName: 'styles__LoggedIn',
  componentId: 'e9mwi9-12'
})(['padding:0 10px;display:flex;align-items:center;']);

var MrAuto = exports.MrAuto = _styledComponents2.default.span.withConfig({
  displayName: 'styles__MrAuto',
  componentId: 'e9mwi9-13'
})(['margin-right:auto !important;']);