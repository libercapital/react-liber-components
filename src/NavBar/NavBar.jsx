import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { SimpleDropDown } from '../DropDown';
import { getScreenSize } from '../Util';
import {
  Nav,
  Content,
  Logo,
  Menu,
  Label,
  Sandwich,
} from './NavBar.styles';

const button = (menu) => {
  const { icon = null, label, sublabel = null } = menu;
  return (
    <Label>
      <li>{icon}{label}</li>
      {sublabel && <li><span>{sublabel}</span></li>}
    </Label>
  );
};

const NavBar = ({
  children, menu, contentSize, window,
}) => {
  const { content } = menu;
  const [screenSize, setScreenSize] = useState(window ? getScreenSize(window) : 'large');

  useEffect(() => {
    if (window) {
      const handleResize = () => {
        setScreenSize(getScreenSize(window));
      };
      window.addEventListener('resize', handleResize);
      return () => { window.removeEventListener('resize', handleResize); };
    }
    return undefined;
  }, window);

  const small = screenSize === 'small' || screenSize === 'medium';
  return (
    <Nav>
      <Content size={contentSize}>
        <Logo />
        {children}
        <Menu>
          <SimpleDropDown
            position="bottom-left"
            label={!small ? button(menu) : undefined}
            customButton={small ? <Sandwich /> : undefined}
          >
            {content}
          </SimpleDropDown>
        </Menu>
      </Content>
    </Nav>
  );
};

NavBar.propTypes = {
  children: PropTypes.oneOfType([PropTypes.any]),
  menu: PropTypes.shape({
    icon: PropTypes.node,
    label: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
    sublabel: PropTypes.string,
    content: PropTypes.oneOfType([PropTypes.any]),
  }),
  contentSize: PropTypes.string,
  window: PropTypes.objectOf(PropTypes.any),
};

NavBar.defaultProps = {
  children: null,
  menu: {},
  contentSize: null,
  window: null,
};

export default NavBar;
