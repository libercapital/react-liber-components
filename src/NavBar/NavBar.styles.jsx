import styled, { css } from 'styled-components';
import { Logo as LiberLogo } from '../Logo';
import { MdMenu } from '../Icons';
import { handleThemeFromObject } from '../Themes';

const getTheme = (value, fallback) => css`
    ${props => handleThemeFromObject(props.theme, value, fallback)}
`;

export const Nav = styled.nav`
  width: 100vw;
  max-width: 100%;
  min-height: 96px;
  max-height: 96px;
  background-color: ${getTheme('colors.white', '#ffffff')};
  margin-bottom: 12px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Content = styled.div`
  height: 100%;
  width: ${props => props.width || '1280px'};
  max-width: 100%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
`;


export const Logo = styled(LiberLogo)`
  --size: 96px;
  min-width: var(--size);
  min-height: var(--size);
  max-width: var(--size);
  max-height: var(--size);
`;
export const Menu = styled.div`
  margin-left: auto;
  margin-right: 32px;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
`;


export const Label = styled.ul`
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  line-height: 22px;
  text-align: right;
  color: #405f71;
  margin: 0px;

  list-style-type: none;

  & > li > svg{
    fill: #798e9b;
  }

  & > li > span {
    font-family: Roboto;
    font-size: 12px;
    line-height: 17px;
    letter-spacing: 0px;
    text-align: right;
    color: #b6c2c9;
  }
`;

export const Sandwich = styled(MdMenu)`
  --size: 32px;
  min-width: var(--size);
  min-height: var(--size);
  max-width: var(--size);
  max-height: var(--size);
  fill: #798e9b;
`;
