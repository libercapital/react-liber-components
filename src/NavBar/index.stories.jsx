import React from 'react';
import styled from 'styled-components';
import NavBar from './NavBar';

const Viewport = styled.div`
  width: 100vw;
  height: 100vh;
  background-color: #fafdff;
`;

export default {
  title: 'Components 2.0|NavBar',

  parameters: {
    jest: ['NavBar'],
  },
};

export const navBar = () => (
  <Viewport>
    <NavBar
      menu={{
        label: 'Nome do usuÃ¡rio',
        sublabel: 'Nome da empresa',
      }}
    />
  </Viewport>
);

navBar.story = {
  name: 'NavBar',
};
