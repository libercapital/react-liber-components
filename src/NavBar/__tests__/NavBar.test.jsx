import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import NavBar from '../NavBar';

describe('Testing NavBar Component', () => {
  let props;
  configure({ adapter: new Adapter() });
  beforeEach(() => {
    props = {
      menu: {
        label: 'label',
        sublabel: 'sublabel',
        icon: <svg />,
      },
      contentSize: '200px',
      children: <span />,
    };
  });
  it('should render correctly with props', () => {
    const wrapper = shallow(<NavBar {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly without props', () => {
    const wrapper = shallow(<NavBar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
