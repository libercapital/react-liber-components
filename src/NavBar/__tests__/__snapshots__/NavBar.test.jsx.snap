// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NavBar Component should render correctly with props 1`] = `
<NavBarstyles__Nav>
  <NavBarstyles__Content
    size="200px"
  >
    <NavBarstyles__Logo />
    <span />
    <NavBarstyles__Menu>
      <SimpleDropDown
        autoPosition={true}
        containerSize={false}
        customButton={null}
        hasOutsideLabel={false}
        label={
          <ForwardRef(NavBarstyles__Label)>
            <li>
              <svg />
              label
            </li>
            <li>
              <span>
                sublabel
              </span>
            </li>
          </ForwardRef(NavBarstyles__Label)>
        }
        open={null}
        position="bottom-left"
        window={[Window]}
      />
    </NavBarstyles__Menu>
  </NavBarstyles__Content>
</NavBarstyles__Nav>
`;

exports[`Testing NavBar Component should render correctly without props 1`] = `
<NavBarstyles__Nav>
  <NavBarstyles__Content
    size={null}
  >
    <NavBarstyles__Logo />
    <NavBarstyles__Menu>
      <SimpleDropDown
        autoPosition={true}
        containerSize={false}
        customButton={null}
        hasOutsideLabel={false}
        label={
          <ForwardRef(NavBarstyles__Label)>
            <li />
          </ForwardRef(NavBarstyles__Label)>
        }
        open={null}
        position="bottom-left"
        window={[Window]}
      />
    </NavBarstyles__Menu>
  </NavBarstyles__Content>
</NavBarstyles__Nav>
`;
